calico_node/startup/startup.go

```diff
@@ -906,6 +913,84 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+type ClientV1 string
+
+func (_ ClientV1) Blah() (string, bool, error) {
+	return "v2.6.1", true, nil
+}
+
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface) error {
+	var cv1 ClientV1
+
+	// Only run this migration if using KDD
+	if cfg.Spec.DatastoreType != apiconfig.Kubernetes {
+		switch v, ver := getClusterVersion(ctx, c, cv1); v {
+		case VUnknown:
```

Comment:
If we hit an error we need to exit, yeah?

Reply:
Are you referring to inside of getClusterVersion? To return an error from in there the Gets used would need to tell us if the Version info was just not present or if there was a problem connecting to the server.

Reply:
That's why I have the TODO below here about making a test connection to the database to know if the problem was in the connection or the data not existing yet.